refactor

nemo0824 opened the pull request:
router router/render 분리 -> 추상화
js 폴더 정리
이벤트 navbar  교체  -> 오류방지

nemo0824 committed:
refactor
router router/render 분리 -> 추상화
js 폴더 정리
이벤트 navbar  교체  -> 오류방지

imjaehoo commented on service/dataService.js:
```diff
@@ -1,5 +1,5 @@
 export async function fetchAllData() {
-  const response = await fetch('/posts.json');
+  const response = await fetch('/mock/posts.json');
```
적어도 이 과제에서 이 데이터는 mock이 아니에요

nemo0824 replied:
넵 mock 데이터의 정의 다시한번 확인해보겠습니다

imjaehoo commented on router/router.js:
```diff
@@ -0,0 +1,61 @@
+}
+
+export function setupRouter() {
+  renderPage(location.pathname);
```
`setupRouter` 함수를 사용하는 쪽에서 함수 내부적으로 페이지 렌더링을 한다는 사실을 기대하거나 예측할 수 있을까요?

imjaehoo commented on router/router.js:
```diff
@@ -0,0 +1,61 @@
+    renderPage(location.pathname);
+  });
+
+  document.addEventListener('click', (e) => {
+    if (e.target.tagName === 'A' && e.target.hasAttribute('data-link')) {
+      e.preventDefault();
+      const url = e.target.getAttribute('href');
+      history.pushState(null, null, url);
+      renderPage(new URL(url, location.origin).pathname);
+    }
+  });
```
이 로직은 `setNavEvent`와 무엇이 다른가요?

imjaehoo commented on router/router.js:
```diff
@@ -0,0 +1,61 @@
+// 렌더링
+async function renderPage(route) {
+  const app = document.getElementById('app');
+  const handler = routes[route];
+  app.innerHTML = await handler();
```
전반적으로 변수 이름 짓기를 많이 고민해보시면 좋겠습니다 :)

imjaehoo commented on router/router.js:
```diff
@@ -0,0 +1,61 @@
+// category
+async function renderCategoryPage(category) {
+  const data = await fetchCategoryData(category);
+  console.log(data);
```
pr에 의미없는 코드는 없어야 합니다.

nemo0824 replied:
네 신경쓰겠습니다!

imjaehoo commented on router/router.js:
```diff
@@ -0,0 +1,61 @@
+  });
+
+  document.addEventListener('click', (e) => {
+    if (e.target.tagName === 'A' && e.target.hasAttribute('data-link')) {
```
class를 쓰지 않고 태그이름&속성을 사용하시는 이유가 있나요?

imjaehoo commented on router/router.js:
```diff
@@ -0,0 +1,61 @@
+  fetchAllData,
+} from '../service/dataservice.js';
+
+const routes = {
```
이 객체가 정말 `routes`인가요?

imjaehoo commented on router/navbar.js:
이 파일 이름은 `setNavEvent`가 적합해보여요

imjaehoo commented on app.js:
```diff
@@ -1,5 +1,5 @@
-import { setupRouter } from './router.js';
-import { setNavEvent } from './navbar.js';
+import { setupRouter } from './router/router.js';
+import { setNavEvent } from './router/navbar.js';
```
지금은 일단 구현만 하고 나중엔 디렉토리 구조도 고민을 해보면 좋을 것 같아요~

nemo0824 committed:
fix
1.navbar 동적으로 관리할수있도록 app 요소안에서 동적으로 렌더링
2.event요소 선택자 기준 관리
3.rotue, mock, 구조 등 네이밍에 맞게 변경
